sdk/cosmos/azure-cosmos-benchmark/src/main/java/com/azure/cosmos/benchmark/linkedin/impl/QueryExecutor.java

```diff
@@ -105,7 +105,14 @@
                 _logger.logDebugInfo(Constants.METHOD_SQL_QUERY, query, activeCollection, _clock.millis() - startTime,
                     activityId, null);
             }
-            return _responseHandler.convertFeedResponse(responseList);
+
+            // Map the response as K-V values
+            final BatchGetResult<K, V> result = _responseHandler.convertFeedResponse(responseList);
+            if (result.getResults().size() == 0) {
+                _metrics.logCounterMetric(Metrics.Type.NOT_FOUND);
```

Comment:
I am not sure we should do NOT_FOUND thing for query , as in regular flow also there are some feed response which can return intermittent 0 results. Also what we information we will infer via this. @mbhaskar  thoughts?

Reply:
This gathers a data point which'd help ensure we are not making requests for non-existent documents. It's similar to the scenario for GET

Reply:
To add, if the feed response is empty, the K-V map in BatchGetResult will also be empty.